lib/dll_loader_helper.ex

```diff
@@ -3,14 +3,30 @@ defmodule DLLLoaderHelper do
   DLL loader helper.
   """
 
+  require Logger
+
+  @on_load :load_nif
+  def load_nif do
+    case :os.type() do
+      {:win32, _} ->
+        nif_file = '#{:code.priv_dir(:dll_loader_helper)}/dll_loader_helper'
+
+        case :erlang.load_nif(nif_file, 0) do
+          :ok -> :ok
+          {:error, {:reload, _}} -> :ok
+          {:error, reason} -> {:error, reason}
+        end
+
+      _ ->
+        :ok
+    end
+  end
+
   @doc """
   Add a directory to DLL search path for Windows.
   """
   @spec addDLLDirectory(binary()) :: :ok | {:error, String.t()}
   def addDLLDirectory(dir) when is_binary(dir) do
-    case :os.type() do
-      {:win32, _} -> :dll_loader_helper_nif.addDLLDirectory(dir)
-      _ -> :ok
-    end
```

Comment:
If we don't load a NIF, then this version will stay in the module, which is exactly what we want for Unix! :)